e621/endpoints/my_endpoints.py

```diff
@@ -1,18 +1,213 @@
+        has_wiki: Optional[bool] = None,
+        has_artist: Optional[bool] = None,
+        limit: Optional[int] = None,
+        page: Optional[str] = None,
+    ) -> List[Tag]:
+        raw_tags = self._api.session.request(
+            "GET",
+            "tags.json",
+            params={
+                "search[name_matches]": name_matches,
+                "search[category]": category,
+                "search[order]": order,
+                "search[hide_empty]": hide_empty,
+                "search[has_wiki]": has_wiki,
+                "search[has_artist]": has_artist,
+                "limit": limit,
+                "page": page,
+            },
+        ).json()
+        return [Tag(**tag) for tag in raw_tags]
 
 
 class TagAliases(BaseEndpoint):
-    pass
+    def get(self, tag_alias_id: int) -> TagAlias:
+        return TagAliases(**self._api.session.request("GET", f"tag_aliases/{tag_alias_id}.json").json())
+
+    def search(
+        self,
+        name_matches: Optional[str] = None,
+        status: Optional[str] = None,
+        order: Optional[str] = None,
+        antecedent_tag_category: Optional[str] = None,
+        consequent_tag_category: Optional[str] = None,
+        limit: Optional[int] = None,
+        page: Optional[str] = None,
+    ) -> List[TagAlias]:
+        raw_tag_aliases = self._api.session.request(
+            "GET",
+            "tag_aliases.json",
+            params={
+                "search[name_matches]": name_matches,
+                "search[status]": status,
+                "search[order]": order,
+                "search[antecedent_tag][category]": antecedent_tag_category,
+                "search[consequent_tag][category]": consequent_tag_category,
+                "limit": limit,
+                "page": page,
+            },
+        ).json()
+        return [TagAlias(**tag_alias) for tag_alias in raw_tag_aliases]
 
 
 class Notes(BaseEndpoint):
-    pass
+    def get(self, note_id: int) -> Note:
+        return Note(**self._api.session.request("GET", f"notes/{note_id}.json").json())
+
+    def search(
+        self,
+        body_matches: Optional[str] = None,
+        post_id: Optional[int] = None,
+        post_tags_match: Optional[str] = None,
+        creator_name: Optional[str] = None,
+        creator_id: Optional[int] = None,
+        is_active: Optional[bool] = None,
+        limit: Optional[int] = None,
+    ) -> List[Note]:
+        raw_notes = self._api.session.request(
+            "GET",
+            "notes.json",
+            params={
+                "search[body_matches]": body_matches,
+                "search[post_id]": post_id,
+                "search[post_tags_match]": post_tags_match,
+                "search[creator_name]": creator_name,
+                "search[creator_id]": creator_id,
+                "search[is_active]": is_active,
+                "limit": limit,
+            },
+        ).json()
+        return [Note(**note) for note in raw_notes]

```

Comment:
You don't handle the case of empty response. You see, their API is weird so it returns an array if there are results and a dict if there is none. 
Here's a rundown of return types in case of >0 results:
Posts
```json
{"posts": [post_info, ...]}
```
Favorites
```json
{"posts": [post_info, ...]}
```
For everything else it will just return the array with the items instead of a dictionary with a single key.


Here's a rundown of return types in case of 0 results:
Posts
`{'posts': []}`
Post_flags
`{'post_flags': []}`
Tags
`{'tags': []}`
Tag_aliases
`{'tag_aliases': []}`
Notes
`{'notes': []}`
Pools
`[]`
Favorites
`{'posts': []}`


Please, look at all your methods and handle them in accordance with this information.